### ISSUE TITLE
Add `commentPattern` option to `no-fallthrough` rule

### ISSUE THREAD

**Issue by alvov** [eslint/eslint#5757]
**What version of ESLint are you using?**
2.5.3
**What rule do you want to change?**
`no-fallthrough`
**What code should be flagged as incorrect with this change?**
```
/*eslint no-fallback: ["error", { "commentPattern": "break[\\s\\w]*omitted" }]*/
switch(foo) {
    case 0: a();
    /* fallthrough */
    default: c();
}
```
**What happens when the rule is applied to this code now?**
Errors:
`Expected a 'break' statement before 'case'.`
`Expected a 'break' statement before 'default'.`

It seems useful to be able to customize the comment pattern for intentional fallthrough (default pattern for now is `/falls?\s?through/i`), especially when one tends to write comments in a language other than english. Consider:
```
/*eslint no-fallback: ["error", { "commentPattern": "break" }]*/
switch (foo) {
    case 0: a();
    // break пропущен умышленно!
    case 1: b();
}
```

**Comment by ilyavolodin** [comment]
Hmm... I thought at some point (a while back) we made a change that would consider any comment as a fallthrough comment.

**Comment by platinumazure** [comment]
@ilyavolodin I took a spin through `no-fallthrough`, `default-case`, `no-empty`, and `no-empty-function`. Here is what I found:

* `no-fallthrough` expects the regex cited in the initial post of this issue
* `default-case` expects a comment to have the value "no default"
* `no-empty` allows any comment
* `no-empty-function` allows any comment

I think `no-fallthrough` and `default-case` should be augmented to support configurable comment values or regexes. (I do think it's probably best to require that a value or regex be specified, to avoid false negatives where a trailing switch comment suppresses the rule even if the user doesn't desire that particular comment to do so.)

This issue, of course, covers `no-fallthrough` primarily, so I will not discuss `default-case` any further here unless we decide to open a new issue.

**Comment by ilyavolodin** [comment]
I guess it makes sense for `no-empty` and `no-empty-function` to accept any comment vs. `no-fallthrough`. I would be OK with adding a new option to both `no-fallthrough` and `no-default-case` to accept RegEx.

**Comment by alvov** [comment]
@platinumazure I think I get your point (though I'm not 100% sure), but in that case the change will be breaking. I mean, existing code with 'fall through' comments will generate errors if the option becomes required. Of course I will fix my PR if thats okay.
@ilyavolodin should I add a new option to `no-default-case` in this PR or create another issue?

**Comment by ilyavolodin** [comment]
@alvov We need input from other members of the team before accepting this issue, but `no-default-case` should be tracked as a separate issue and PR

**Comment by nzakas** [comment]
@eslint/eslint-team thoughts on this?

**Comment by platinumazure** [comment]
@alvov I can't see why this would be a breaking change. If the rule option is omitted, we can use the same regex we're currently using as a fallback.

@nzakas For what it's worth, I would champion this on usability/i18n grounds.

**Comment by platinumazure** [comment]
@alvov And I see where I went wrong in my earlier comment. When I said I wanted to require a regular expression, I meant that I don't want the rule to accept any comment as a fall-through indicator. Rather, the user should provide a regular expression or we should use the default. Sorry for being unclear.

**Comment by BYK** [comment]
I'm :+1: for what @ilyavolodin and @platinumazure says

**Comment by nzakas** [comment]
Okay, we will use this issue to track `no-fallthrough`. Please open a separate issue for `no-default-case`.
